core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -741,4 +782,39 @@ public Executor getPersistenceExecutor() {
    public StreamingMarshaller getMarshaller() {
       return m;
    }
+
+   private void removeCacheLoader(String storeType, Collection<CacheLoader> collection) {
+      Iterator<CacheLoader> clIt = collection.iterator();
```

Comment:
We can actually do `removeIf`.  This way the implementation can decided best way to iterate.